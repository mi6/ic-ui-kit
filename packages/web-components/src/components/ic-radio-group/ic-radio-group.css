@import "../../global/normalize.css";

:host {
  display: block;
}

ic-input-label.error {
  color: var(--ic-status-error);
}

ic-input-validation {
  margin-top: var(--ic-space-sm);
}

:host(.small) ic-input-validation {
  margin-top: calc(var(--ic-space-sm) / 2);
}

ic-input-label ic-typography {
  margin-bottom: var(--ic-space-sm);
}

:host(.small) ic-input-label ic-typography {
  margin-bottom: calc(var(--ic-space-sm) / 2);
}

.radio-buttons-container {
  display: flex;
  flex-direction: column;
  gap: var(--ic-space-xxs);
}

:host(.small) .radio-buttons-container {
  gap: var(--ic-space-xxxs);
}

:host([orientation="horizontal"]) .radio-buttons-container {
  display: flex;
  flex-direction: row;
  gap: calc(var(--ic-space-xl) + var(--ic-space-xs));
}

:host(.small[orientation="horizontal"]) .radio-buttons-container {
  gap: var(--ic-space-xl);
}
