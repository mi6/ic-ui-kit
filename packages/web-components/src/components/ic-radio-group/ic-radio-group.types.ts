export interface IcChangeEventDetail {
  value: string;
  selectedOption: {
    radio: HTMLIcRadioOptionElement;
    textFieldValue?: string;
  };
}
