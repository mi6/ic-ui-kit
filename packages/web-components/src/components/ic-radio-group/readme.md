# ic-radio-group



<!-- Auto Generated Below -->


## Properties

| Property             | Attribute           | Description                                                                                                                                                                                                                      | Type                                      | Default      |
| -------------------- | ------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------- | ------------ |
| `disabled`           | `disabled`          | If `true`, the disabled state will be set.                                                                                                                                                                                       | `boolean`                                 | `false`      |
| `helperText`         | `helper-text`       | The helper text that will be displayed for additional field guidance.                                                                                                                                                            | `string`                                  | `undefined`  |
| `hideLabel`          | `hide-label`        | If `true`, the label will be hidden and the required label value will be applied as an aria-label.                                                                                                                               | `boolean`                                 | `false`      |
| `label` _(required)_ | `label`             | The label for the radio group to be displayed.                                                                                                                                                                                   | `string`                                  | `undefined`  |
| `name` _(required)_  | `name`              | The name for the radio group to differentiate from other groups.                                                                                                                                                                 | `string`                                  | `undefined`  |
| `orientation`        | `orientation`       | The orientation of the radio buttons in the radio group. If there are more than two radio buttons in a radio group or either of the radio buttons use the `additional-field` slot, then the orientation will always be vertical. | `"horizontal" \| "vertical"`              | `"vertical"` |
| `required`           | `required`          | If `true`, the radio group will require a value.                                                                                                                                                                                 | `boolean`                                 | `false`      |
| `small`              | `small`             | If `true`, the small styling will be applied to the radio group.                                                                                                                                                                 | `boolean`                                 | `false`      |
| `validationStatus`   | `validation-status` | The validation status - e.g. 'error' \| 'warning' \| 'success'.                                                                                                                                                                  | `"" \| "error" \| "success" \| "warning"` | `""`         |
| `validationText`     | `validation-text`   | The validation text - e.g. 'error' \| 'warning' \| 'success'.                                                                                                                                                                    | `string`                                  | `""`         |


## Events

| Event      | Description                          | Type                               |
| ---------- | ------------------------------------ | ---------------------------------- |
| `icChange` | Emitted when a user selects a radio. | `CustomEvent<IcChangeEventDetail>` |


## Dependencies

### Depends on

- [ic-input-label](../ic-input-label)
- [ic-input-validation](../ic-input-validation)

### Graph
```mermaid
graph TD;
  ic-radio-group --> ic-input-label
  ic-radio-group --> ic-input-validation
  ic-input-label --> ic-typography
  ic-input-validation --> ic-typography
  style ic-radio-group fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


