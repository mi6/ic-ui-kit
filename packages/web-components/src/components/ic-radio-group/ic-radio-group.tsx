import {
  Component,
  Host,
  h,
  Prop,
  State,
  Listen,
  Element,
  Event,
  EventEmitter,
} from "@stencil/core";
import {
  hasValidationStatus,
  isSlotUsed,
  onComponentRequiredPropUndefined,
  removeDisabledFalse,
  renderHiddenInput,
} from "../../utils/helpers";
import {
  IcInformationStatusOrEmpty,
  IcOrientation,
  IcValueEventDetail,
} from "../../utils/types";
import { IcChangeEventDetail } from "./ic-radio-group.types";

@Component({
  tag: "ic-radio-group",
  styleUrl: "ic-radio-group.css",
  shadow: true,
})
export class RadioGroup {
  private radioOptions: HTMLIcRadioOptionElement[];

  @Element() host: HTMLIcRadioGroupElement;

  @State() checkedValue: string = "";
  @State() selectedChild: number = -1;

  /**
   * If `true`, the disabled state will be set.
   */
  @Prop() disabled: boolean = false;

  /**
   * The helper text that will be displayed for additional field guidance.
   */
  @Prop() helperText: string;

  /**
   * If `true`, the label will be hidden and the required label value will be applied as an aria-label.
   */
  @Prop() hideLabel: boolean = false;

  /**
   * The label for the radio group to be displayed.
   */
  @Prop() label!: string;

  /**
   * The name for the radio group to differentiate from other groups.
   */
  @Prop() name!: string;

  /**
   * The orientation of the radio buttons in the radio group. If there are more than two radio buttons in a radio group or either of the radio buttons use the `additional-field` slot, then the orientation will always be vertical.
   */
  @Prop({ reflect: true, mutable: true }) orientation: IcOrientation =
    "vertical";

  /**
   * If `true`, the radio group will require a value.
   */
  @Prop() required: boolean = false;

  /**
   * If `true`, the small styling will be applied to the radio group.
   */
  @Prop() small: boolean = false;
  /**
   * The validation status - e.g. 'error' | 'warning' | 'success'.
   */
  @Prop() validationStatus: IcInformationStatusOrEmpty = "";
  /**
   * The validation text - e.g. 'error' | 'warning' | 'success'.
   */
  @Prop() validationText: string = "";

  /**
   * Emitted when a user selects a radio.
   */
  @Event() icChange: EventEmitter<IcChangeEventDetail>;

  componentWillLoad(): void {
    removeDisabledFalse(this.disabled, this.host);
  }

  componentDidLoad(): void {
    this.radioOptions = Array.from(
      this.host.querySelectorAll("ic-radio-option")
    );

    this.radioOptions.forEach((radioOption, index) => {
      if (!radioOption.selected) {
        radioOption.selected = this.checkedValue === radioOption.value;
      }
      radioOption.name = this.name;
      radioOption.groupLabel = this.label;
      if (radioOption.selected) {
        this.selectedChild = index;
        this.checkedValue = radioOption.value;
      }
    });
    this.radioOptions[0].shadowRoot.querySelector("input").tabIndex =
      this.selectedChild > 0 ? -1 : 0;

    if (
      this.orientation === "horizontal" &&
      this.radioOptions !== undefined &&
      (this.radioOptions.length > 2 ||
        (this.radioOptions.length === 2 &&
          (isSlotUsed(this.radioOptions[0], "additional-field") ||
            isSlotUsed(this.radioOptions[1], "additional-field"))))
    ) {
      this.orientation = "vertical";
    }

    onComponentRequiredPropUndefined(
      [
        { prop: this.label, propName: "label" },
        { prop: this.name, propName: "name" },
      ],
      "Radio Group"
    );
  }

  @Listen("icCheck")
  selectHandler(event: CustomEvent<IcValueEventDetail>): void {
    this.checkedValue = event.detail.value;
    const selectedOption = event.target as HTMLIcRadioOptionElement;
    this.icChange.emit({
      value: this.checkedValue,
      selectedOption: {
        radio: selectedOption,
        textFieldValue: selectedOption?.querySelector("ic-text-field")?.value,
      },
    });

    if (this.radioOptions !== undefined) {
      this.radioOptions.forEach((radioOption, index) => {
        radioOption.selected = this.checkedValue === radioOption.value;
        if (radioOption.selected) {
          this.selectedChild = index;
        }
      });
      this.radioOptions[0].shadowRoot.querySelector("input").tabIndex =
        this.selectedChild > 0 ? -1 : 0;
    }
  }

  @Listen("icSelectedChange")
  changeHandler(): void {
    const selectedOption = this.radioOptions.findIndex(
      (radioOption) => radioOption.selected
    );
    if (selectedOption < 0) {
      this.radioOptions[0].shadowRoot.querySelector("input").tabIndex = 0;
      this.selectedChild = selectedOption;
    }
  }

  private handleKeyDown = (event: KeyboardEvent): void => {
    switch (event.key) {
      case "ArrowDown":
      case "ArrowRight":
        this.radioOptions[
          this.getNextItemToSelect(this.selectedChild, true)
        ].click();
        event.preventDefault();
        break;
      case "ArrowUp":
      case "ArrowLeft":
        this.radioOptions[
          this.getNextItemToSelect(this.selectedChild, false)
        ].click();
        event.preventDefault();
    }
  };

  private getNextItemToSelect = (
    currentItem: number,
    movingDown: boolean
  ): number => {
    const numRadios = this.radioOptions.length - 1;

    if (currentItem < 1) {
      currentItem = 0;
    }

    let nextItem = movingDown ? currentItem + 1 : currentItem - 1;

    //check if wrap around necessary
    if (nextItem < 0) {
      nextItem = numRadios;
    } else if (nextItem > numRadios) {
      nextItem = 0;
    }

    //if next item is disabled then find next
    if (this.radioOptions[nextItem].disabled) {
      nextItem = this.getNextItemToSelect(nextItem, movingDown);
    }

    return nextItem;
  };

  render() {
    renderHiddenInput(
      true,
      this.host,
      this.name,
      this.checkedValue,
      this.disabled
    );

    return (
      <Host onKeyDown={this.handleKeyDown} class={{ small: this.small }}>
        <div
          role="radiogroup"
          aria-label={`${this.label}${this.required ? ", required" : ""}`}
        >
          {!this.hideLabel && (
            <ic-input-label
              class={{ [`${this.validationStatus}`]: true }}
              label={this.label}
              helperText={this.helperText}
              required={this.required}
              disabled={this.disabled}
            ></ic-input-label>
          )}
          <div class="radio-buttons-container">
            <slot></slot>
          </div>
        </div>
        {hasValidationStatus(this.validationStatus, this.disabled) && (
          <ic-input-validation
            ariaLiveMode="polite"
            status={this.validationStatus}
            message={this.validationText}
          ></ic-input-validation>
        )}
      </Host>
    );
  }
}
