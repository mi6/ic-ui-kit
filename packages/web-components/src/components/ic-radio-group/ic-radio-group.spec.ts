import { newSpecPage } from "@stencil/core/testing";
import { RadioGroup } from "./ic-radio-group";
import { RadioOption } from "../ic-radio-option/ic-radio-option";

describe("ic-radio-group", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test">
        <ic-radio-option value="test"></ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot("renders");
  });

  it("should render as required", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test"></ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-required");
  });

  it("should render as helper text", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" helperText="helper test">
        <ic-radio-option value="test"></ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-helpertext");
  });

  it("should render with selected option", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-selected-option");
  });

  it("should render with validation status", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required validation-status="error" validation-text="error">
        <ic-radio-option value="test" selected></ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-validation-status");
  });

  it("should render radio option disabled", async () => {
    const page = await newSpecPage({
      components: [RadioOption],
      html: `<ic-radio-group label="test label" name="test">
        <ic-radio-option value="test" disabled label="test label" group-label="test group"></ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-disabled");
  });

  it("should render with unselected static additional field", async () => {
    const page = await newSpecPage({
      components: [RadioOption],
      html: `<ic-radio-group label="test label" name="test">
        <ic-radio-option value="test" disabled label="test label" group-label="test group">
         <ic-text-field slot="additional-field" placeholder="Placeholder" label="Test label"></ic-text-field>
        </ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot(
      "renders-with-unselected-additional-field"
    );
  });

  it("should render with selected static additional field", async () => {
    const page = await newSpecPage({
      components: [RadioOption],
      html: `<ic-radio-group label="test label" name="test">
        <ic-radio-option value="test" selected label="test label" group-label="test group">
         <ic-text-field slot="additional-field" placeholder="Placeholder" label="Test label"></ic-text-field>
        </ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-selected-additional-field");
  });

  it("should render with dynamic additional field", async () => {
    const page = await newSpecPage({
      components: [RadioOption],
      html: `<ic-radio-group label="test label" name="test">
        <ic-radio-option value="test" label="test label" group-label="test group" additional-field-display="dynamic" selected>
         <ic-text-field slot="additional-field" placeholder="Placeholder" label="Test label"></ic-text-field>
        </ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.root).toMatchSnapshot("renders-with-dynamic-additional-field");
  });

  it("should emit an updated checked value when selected", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>    
      </ic-radio-group>`,
    });

    const callbackFn = jest.fn();
    page.doc.addEventListener("icChange", callbackFn);
    page.rootInstance.selectHandler({ detail: { value: "true" } });
    await page.waitForChanges();

    expect(callbackFn).toHaveBeenCalled();
  });

  it("should get the next item to select", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>   
        <ic-radio-option value="test2"></ic-radio-option>   
      </ic-radio-group>`,
    });

    const nextItemCalc = page.rootInstance.getNextItemToSelect(0, true);

    page.waitForChanges();
    expect(nextItemCalc).toBe(1);
  });

  it("should skip getting the next item when it's disabled", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>   
        <ic-radio-option value="test2" disabled></ic-radio-option> 
        <ic-radio-option value="test3"></ic-radio-option>    
      </ic-radio-group>`,
    });

    const nextItemCalc = page.rootInstance.getNextItemToSelect(0, true);
    page.waitForChanges();
    expect(nextItemCalc).toBe(2);
  });

  it("should not move onto the next item when nextItem < 0", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>    
      </ic-radio-group>`,
    });

    const nextItemCalc = page.rootInstance.getNextItemToSelect(0, false);

    page.waitForChanges();
    expect(nextItemCalc).toBe(0);
  });

  it("should select the first radio when nextItem > numRadios", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>   
        <ic-radio-option value="test"></ic-radio-option>  
        <ic-radio-option value="test"></ic-radio-option>   
      </ic-radio-group>`,
    });

    const nextItemCalc = page.rootInstance.getNextItemToSelect(2, true);

    page.waitForChanges();
    expect(nextItemCalc).toBe(0);
  });

  it("should select the radio option when clicked", async () => {
    const page = await newSpecPage({
      components: [RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>    
      </ic-radio-group>`,
    });

    const callbackFn = jest.fn();
    page.doc.addEventListener("icCheck", callbackFn);
    page.rootInstance.selectedChangeHandler("true");
    await page.waitForChanges();

    expect(callbackFn).toHaveBeenCalled();
  });

  it("should emit new radio value when text field value given", async () => {
    const page = await newSpecPage({
      components: [RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>    
      </ic-radio-group>`,
    });

    const callbackFn = jest.fn();
    page.doc.addEventListener("icCheck", callbackFn);
    page.rootInstance.textfieldValueHandler({ detail: { value: "value" } });
    await page.waitForChanges();

    expect(callbackFn).toHaveBeenCalled();
  });

  it("should emit default radio value when no text field value given", async () => {
    const page = await newSpecPage({
      components: [RadioOption],
      html: `<ic-radio-group label="test label" name="test" required>
        <ic-radio-option value="test" selected></ic-radio-option>    
      </ic-radio-group>`,
    });

    const callbackFn = jest.fn();
    page.doc.addEventListener("icCheck", callbackFn);
    page.rootInstance.textfieldValueHandler({ detail: { value: "" } });
    await page.waitForChanges();

    expect(callbackFn).toHaveBeenCalled();
  });

  it("should test form reset event", async () => {
    const page = await newSpecPage({
      components: [RadioOption],
      html: `<form>
        <ic-radio-option value="test" selected></ic-radio-option>    
        <button id="resetButton" type="reset">Reset</button>  
      </form>`,
    });

    expect(page.root.selected).toBe(true);

    page.root.checked = false;
    await page.waitForChanges();

    expect(page.root.checked).toBe(false);

    await page.rootInstance.handleFormReset();
    await page.waitForChanges();

    expect(page.root.selected).toBe(true);

    //test disconnected callback
    page.setContent("");
  });

  it("should change the orientation of the radio group to vertical if the user sets the orientation as horizontal and there are more than 2 radio options in the radio group", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required orientation="horizontal">
        <ic-radio-option value="test1" selected></ic-radio-option>
        <ic-radio-option value="test2" ></ic-radio-option> 
        <ic-radio-option value="test3"></ic-radio-option>    
      </ic-radio-group>`,
    });

    expect(page.rootInstance.orientation).toMatch("vertical");
  });

  it("should change the orientation of the radio group to vertical if the user has additional fields on any of the radio buttons in the group", async () => {
    const page = await newSpecPage({
      components: [RadioGroup, RadioOption],
      html: `<ic-radio-group label="test label" name="test" required orientation="horizontal">
        <ic-radio-option value="test1" selected></ic-radio-option>
        <ic-radio-option value="test" disabled label="test label" group-label="test group">
         <ic-text-field slot="additional-field" placeholder="Placeholder" label="Test label"></ic-text-field>
        </ic-radio-option>      
      </ic-radio-group>`,
    });

    expect(page.rootInstance.orientation).toMatch("vertical");
  });
});
