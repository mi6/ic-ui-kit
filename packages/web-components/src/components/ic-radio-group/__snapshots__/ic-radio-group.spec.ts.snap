// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-radio-group should render as helper text: renders-with-helpertext 1`] = `
<ic-radio-group helpertext="helper test" label="test label" name="test" orientation="vertical">
  <mock:shadow-root>
    <div aria-label="test label" role="radiogroup">
      <ic-input-label label="test label"></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="static" value="test">
    <mock:shadow-root>
      <div class="container">
        <div>
          <input id="ic-radio-option-test-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test-test label"></label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="">
</ic-radio-group>
`;

exports[`ic-radio-group should render as required: renders-required 1`] = `
<ic-radio-group label="test label" name="test" orientation="vertical" required="">
  <mock:shadow-root>
    <div aria-label="test label, required" role="radiogroup">
      <ic-input-label label="test label" required=""></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="static" value="test">
    <mock:shadow-root>
      <div class="container">
        <div>
          <input id="ic-radio-option-test-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test-test label"></label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="">
</ic-radio-group>
`;

exports[`ic-radio-group should render radio option disabled: renders-disabled 1`] = `
<ic-radio-group label="test label" name="test" orientation="vertical">
  <mock:shadow-root>
    <div aria-label="test label" role="radiogroup">
      <ic-input-label label="test label"></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="static" class="disabled" disabled="" group-label="test group" label="test label" value="test">
    <mock:shadow-root>
      <div class="container disabled">
        <div>
          <input disabled="" id="ic-radio-option-test label-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test label-test label">
            test label
          </label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="">
</ic-radio-group>
`;

exports[`ic-radio-group should render with dynamic additional field: renders-with-dynamic-additional-field 1`] = `
<ic-radio-group label="test label" name="test" orientation="vertical">
  <mock:shadow-root>
    <div aria-label="test label" role="radiogroup">
      <ic-input-label label="test label"></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="dynamic" group-label="test group" label="test label" selected="" value="test">
    <mock:shadow-root>
      <div class="container">
        <div>
          <input checked="" id="ic-radio-option-test label-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test label-test label">
            test label
          </label>
        </ic-typography>
      </div>
      <div class="dynamic-container">
        <div class="branch-corner"></div>
        <div>
          <ic-typography variant="caption">
            <p class="dynamic-text">
              This selection requires additional answers
            </p>
          </ic-typography>
          <div>
            <slot name="additional-field"></slot>
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <ic-text-field label="Test label" placeholder="Placeholder" slot="additional-field" value="">
      <mock:shadow-root>
        <ic-input-container>
          <ic-input-label for="ic-text-field-input-2" helpertext="" label="Test label"></ic-input-label>
          <ic-input-component-container validationstatus="">
            <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-2" inputmode="text" name="ic-text-field-input-2" placeholder="Placeholder" type="text" value="">
          </ic-input-component-container>
        </ic-input-container>
      </mock:shadow-root>
    </ic-text-field>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="test">
</ic-radio-group>
`;

exports[`ic-radio-group should render with selected option: renders-with-selected-option 1`] = `
<ic-radio-group label="test label" name="test" orientation="vertical" required="">
  <mock:shadow-root>
    <div aria-label="test label, required" role="radiogroup">
      <ic-input-label label="test label" required=""></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="static" selected="" value="test">
    <mock:shadow-root>
      <div class="container">
        <div>
          <input checked="" id="ic-radio-option-test-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test-test label"></label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="test">
</ic-radio-group>
`;

exports[`ic-radio-group should render with selected static additional field: renders-with-selected-additional-field 1`] = `
<ic-radio-group label="test label" name="test" orientation="vertical">
  <mock:shadow-root>
    <div aria-label="test label" role="radiogroup">
      <ic-input-label label="test label"></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="static" group-label="test group" label="test label" selected="" value="test">
    <mock:shadow-root>
      <div class="container">
        <div>
          <input checked="" id="ic-radio-option-test label-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test label-test label">
            test label
          </label>
        </ic-typography>
      </div>
      <div class="dynamic-container">
        <div>
          <div class="additional-field-wrapper">
            <slot name="additional-field"></slot>
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <ic-text-field label="Test label" placeholder="Placeholder" slot="additional-field" value="">
      <mock:shadow-root>
        <ic-input-container>
          <ic-input-label for="ic-text-field-input-1" helpertext="" label="Test label"></ic-input-label>
          <ic-input-component-container validationstatus="">
            <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" id="ic-text-field-input-1" inputmode="text" name="ic-text-field-input-1" placeholder="Placeholder" type="text" value="">
          </ic-input-component-container>
        </ic-input-container>
      </mock:shadow-root>
    </ic-text-field>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="test">
</ic-radio-group>
`;

exports[`ic-radio-group should render with unselected static additional field: renders-with-unselected-additional-field 1`] = `
<ic-radio-group label="test label" name="test" orientation="vertical">
  <mock:shadow-root>
    <div aria-label="test label" role="radiogroup">
      <ic-input-label label="test label"></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="static" class="disabled" disabled="" group-label="test group" label="test label" value="test">
    <mock:shadow-root>
      <div class="container disabled">
        <div>
          <input disabled="" id="ic-radio-option-test label-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test label-test label">
            test label
          </label>
        </ic-typography>
      </div>
      <div class="dynamic-container">
        <div>
          <div class="additional-field-wrapper">
            <slot name="additional-field"></slot>
          </div>
        </div>
      </div>
    </mock:shadow-root>
    <ic-text-field disabled="" label="Test label" placeholder="Placeholder" slot="additional-field" value="">
      <mock:shadow-root>
        <ic-input-container disabled="">
          <ic-input-label disabled="" for="ic-text-field-input-0" helpertext="" label="Test label"></ic-input-label>
          <ic-input-component-container disabled="" validationstatus="">
            <input aria-describedby="" aria-invalid="false" aria-label="Test label" autocapitalize="off" autocomplete="off" disabled="" id="ic-text-field-input-0" inputmode="text" name="ic-text-field-input-0" placeholder="" type="text" value="">
          </ic-input-component-container>
        </ic-input-container>
      </mock:shadow-root>
    </ic-text-field>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="">
</ic-radio-group>
`;

exports[`ic-radio-group should render with validation status: renders-with-validation-status 1`] = `
<ic-radio-group label="test label" name="test" orientation="vertical" required="" validation-status="error" validation-text="error">
  <mock:shadow-root>
    <div aria-label="test label, required" role="radiogroup">
      <ic-input-label class="error" label="test label" required=""></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
    <ic-input-validation arialivemode="polite" message="error" status="error"></ic-input-validation>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="static" selected="" value="test">
    <mock:shadow-root>
      <div class="container">
        <div>
          <input checked="" id="ic-radio-option-test-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test-test label"></label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="test">
</ic-radio-group>
`;

exports[`ic-radio-group should render: renders 1`] = `
<ic-radio-group label="test label" name="test" orientation="vertical">
  <mock:shadow-root>
    <div aria-label="test label" role="radiogroup">
      <ic-input-label label="test label"></ic-input-label>
      <div class="radio-buttons-container">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <ic-radio-option additional-field-display="static" value="test">
    <mock:shadow-root>
      <div class="container">
        <div>
          <input id="ic-radio-option-test-test label" name="test" role="radio" tabindex="0" type="radio" value="test">
          <span class="checkmark"></span>
        </div>
        <ic-typography class="radio-label" variant="body">
          <label htmlfor="ic-radio-option-test-test label"></label>
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-radio-option>
  <input class="ic-input" name="test" type="hidden" value="">
</ic-radio-group>
`;
